python/ray/util/client/dataclient.py

```diff
@@ -118,8 +117,9 @@ def _blocking_send(self, req: ray_client_pb2.DataRequest
                 lambda: req_id in self.ready_data or self._in_shutdown)
             if self._in_shutdown:
                 raise ConnectionError(
-                    "Cannot send request due to data channel "
-                    f"shutting down. Request: {req}")
+                    "Sending request failed because the data channel "
+                    "is no longer running. This is usually due to an error "
+                    f"during the connection process: {req}")
```

Comment:
The exception is actually just printed immediately!

Reply:
The "Got Error from data channel -- shutting down:" is from the `logger.exception` in dataclient!
```
In [1]: import ray; ray.client("localhost:10001").namespace("abc").connect()
Got Error from data channel -- shutting down:
Traceback (most recent call last):
  File "/home/ubuntu/ray/python/ray/util/client/dataclient.py", line 69, in _data_main
    for response in resp_stream:
  File "/home/ubuntu/anaconda3/envs/ray/lib/python3.7/site-packages/grpc/_channel.py", line 416, in __next__
    return self._next()
  File "/home/ubuntu/anaconda3/envs/ray/lib/python3.7/site-packages/grpc/_channel.py", line 803, in _next
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Exception iterating responses: BLEW UP"
        debug_error_string = "{"created":"@1622666798.655286554","description":"Error received from peer ipv6:[::1]:10001","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"Exception iterating responses: BLEW UP","grpc_status":2}"
>
---------------------------------------------------------------------------
ConnectionError                           Traceback (most recent call last)
<ipython-input-1-2c7379d59651> in <module>
----> 1 import ray; ray.client("localhost:10001").namespace("abc").connect()

~/ray/python/ray/client_builder.py in connect(self)
     95         """
     96         client_info_dict = ray.util.client_connect.connect(
---> 97             self.address, job_config=self._job_config)
     98         dashboard_url = ray.get(
     99             ray.remote(ray.worker.get_dashboard_url).remote())

~/ray/python/ray/util/client_connect.py in connect(conn_str, secure, metadata, connection_retries, job_config, namespace, ignore_version)
     34         connection_retries=connection_retries,
     35         namespace=namespace,
---> 36         ignore_version=ignore_version)
     37 
     38 

~/ray/python/ray/util/client/__init__.py in connect(self, conn_str, job_config, secure, metadata, connection_retries, namespace, ignore_version)
     75                 connection_retries=connection_retries)
     76             self.api.worker = self.client_worker
---> 77             self.client_worker._server_init(job_config)
     78             conn_info = self.client_worker.connection_info()
     79             self._check_versions(conn_info, ignore_version)

~/ray/python/ray/util/client/worker.py in _server_init(self, job_config)
    475                 runtime_env.PKG_DIR = old_dir
    476                 prep_req = ray_client_pb2.PrepRuntimeEnvRequest()
--> 477                 self.data_client.PrepRuntimeEnv(prep_req)
    478         except grpc.RpcError as e:
    479             raise decode_exception(e.details())

~/ray/python/ray/util/client/dataclient.py in PrepRuntimeEnv(self, request, context)
    144                        context=None) -> ray_client_pb2.PrepRuntimeEnvResponse:
    145         datareq = ray_client_pb2.DataRequest(prep_runtime_env=request, )
--> 146         resp = self._blocking_send(datareq)
    147         return resp.prep_runtime_env
    148 

~/ray/python/ray/util/client/dataclient.py in _blocking_send(self, req)
    118             if self._in_shutdown:
    119                 raise ConnectionError(
--> 120                     "Sending request failed because the data channel "
    121                     "terminated. This is usually due to an error "
    122                     f"in handling the most recent request: {req}")

ConnectionError: Sending request failed because the data channel terminated. This is usually due to an error in handling the most recent request: req_id: 2
prep_runtime_env {
}
```